src/um/devicetopology.rs

```diff
@@ -0,0 +1,466 @@
+    GenLocation: EPcxGenLocation,
+    PortConnection: EPxcPortConnection,
+    IsConnected: BOOL,
+}}
+pub type PKSJACK_DESCRIPTION = *mut KSJACK_DESCRIPTION;
+STRUCT!{struct LUID {
+    LowPart: DWORD,
+    HighPart: LONG,
+}}
+pub type PLUID = *mut LUID;
+ENUM!{enum KSJACK_SINK_CONNECTIONTYPE {
+    KSJACK_SINK_CONNECTIONTYPE_HDMI = 0,
+    KSJACK_SINK_CONNECTIONTYPE_DISPLAYPORT = 1,
+}}
+STRUCT!{struct KSJACK_SINK_INFORMATION {
+    ConnType: KSJACK_SINK_CONNECTIONTYPE,
+    ManufacturerId: WORD,
+    ProductId: WORD,
+    AudioLatency: WORD,
+    HDCPCapable: BOOL,
+    AICapable: BOOL,
+    SinkDescriptionLength: UCHAR,
+    SinkDescription: [WCHAR; 32],
+    PortId: LUID,
+}}
+STRUCT!{struct KSJACK_DESCRIPTION2 {
+    DeviceStateInfo: DWORD,
+    JackCapabilities: DWORD,
+}}
+pub type PKSJACK_DESCRIPTION2 = *mut KSJACK_DESCRIPTION2;
+ENUM!{enum DataFlow {
+    In = 0,
+    Out = 1,
+}}
+ENUM!{enum PartType {
+    Connector = 0,
+    Subunit = 1,
+}}
+ENUM!{enum ConnectorType {
+    Unknown_Connector = 0,
+    Physical_Internal = 1,
+    Physical_External = 2,
+    Software_IO = 3,
+    Software_Fixed = 4,
+    Network = 5,
+}}
+RIDL!{#[uuid(0x28f54685, 0x06fd, 0x11d2, 0xb2, 0x7a, 0x00, 0xa0, 0xc9, 0x22, 0x31, 0x96)]
+interface IKsControl(IKsControlVtbl): IUnknown(IUnknownVtbl) {
+    fn KsProperty(
+        Property: PKSPROPERTY,
+        PropertyLength: ULONG,
+        PropertyData: *mut c_void,
+        DataLength: ULONG,
+        BytesReturned: *mut ULONG,
+    ) -> HRESULT,
+    fn KsMethod(
+        Method: PKSMETHOD,
+        MethodLength: ULONG,
+        MethodData: *mut c_void,
+        DataLength: ULONG,
+        BytesReturned: *mut ULONG,
+    ) -> HRESULT,
+    fn KsEvent(
+        Event: PKSEVENT,
+        EventLength: ULONG,
+        EventData: *mut c_void,
+        DataLength: ULONG,
+        BytesReturned: *mut ULONG,
+    ) -> HRESULT,
+}}
+RIDL!{#[uuid(0xc2f8e001, 0xf205, 0x4bc9, 0x99, 0xbc, 0xc1, 0x3b, 0x1e, 0x04, 0x8c, 0xcb)]
+interface IPerChannelDbLevel(IPerChannelDbLevelVtbl): IUnknown(IUnknownVtbl) {
+    fn GetChannelCount(
+        pcChannels: *mut UINT,
+    ) -> HRESULT,
+    fn GetLevelRange(
+        nChannel: UINT,
+        pfMinLevelDB: *mut c_float,
+        pfMaxLevelDB: *mut c_float,
+        pfStepping: *mut c_float,
+    ) -> HRESULT,
+    fn GetLevel(
+        nChannel: UINT,
+        pfLevelDB: *mut c_float,
+    ) -> HRESULT,
+    fn SetLevel(
+        nChannel: UINT,
+        fLevelDB: c_float,
+        pguidEventContext: LPCGUID,
+    ) -> HRESULT,
+    fn SetLevelUniform(
+        fLevelDB: c_float,
+        pguidEventContext: LPCGUID,
+    ) -> HRESULT,
+    fn SetLevelAllChannels(
+        aLevelsDB: *mut c_float,
+        cChannels: ULONG,
+        pguidEventContext: LPCGUID,
+    ) -> HRESULT,
+}}
+RIDL!{#[uuid(0x7fb7b48f, 0x531d, 0x44a2, 0xbc, 0xb3, 0x5a, 0xd5, 0xa1, 0x34, 0xb3, 0xdc)]
+interface IAudioVolumeLevel(IAudioVolumeLevelVtbl): IPerChannelDbLevel(IPerChannelDbLevelVtbl) {}
+}
```

Comment:
This closing brace should be on the previous line.
`interface IAudioVolumeLevel(IAudioVolumeLevelVtbl): IPerChannelDbLevel(IPerChannelDbLevelVtbl) {}}`